iib/workers/tasks/build_containerized_fbc_operations.py

```diff
@@ -0,0 +1,330 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+import logging
+import os
+import tempfile
+from typing import Dict, List, Optional, Set
+
+from iib.common.common_utils import get_binary_versions
+from iib.common.tracing import instrument_tracing
+from iib.exceptions import IIBError
+from iib.workers.api_utils import set_request_state
+from iib.workers.tasks.build import (
+    _cleanup,
+    _update_index_image_build_state,
+    _update_index_image_pull_spec,
+    _skopeo_copy,
+)
+from iib.workers.tasks.celery import app
+from iib.workers.tasks.containerized_utils import (
+    pull_index_db_artifact,
+    write_build_metadata,
+    get_list_of_output_pullspec,
+    cleanup_on_failure,
+    push_index_db_artifact,
+)
+from iib.workers.tasks.git_utils import (
+    create_mr,
+    clone_git_repo,
+    get_git_token,
+    get_last_commit_sha,
+    resolve_git_url,
+    commit_and_push,
+    close_mr,
+)
+from iib.workers.tasks.konflux_utils import (
+    wait_for_pipeline_completion,
+    find_pipelinerun,
+    get_pipelinerun_image_url,
+)
+from iib.workers.tasks.opm_operations import (
+    Opm,
+    opm_registry_add_fbc_fragment_containerized,
+)
+from iib.workers.tasks.utils import (
+    get_resolved_image,
+    prepare_request_for_build,
+    request_logger,
+    set_registry_token,
+    RequestConfigFBCOperation,
+)
+
+__all__ = ['handle_containerized_fbc_operation_request']
+
+log = logging.getLogger(__name__)
+
+
+@app.task
+@request_logger
+@instrument_tracing(
+    span_name="workers.tasks.build.handle_containerized_fbc_operation_request",
+    attributes=get_binary_versions(),
+)
+def handle_containerized_fbc_operation_request(
+    request_id: int,
+    fbc_fragments: List[str],
+    from_index: str,
+    binary_image: Optional[str] = None,
+    distribution_scope: str = '',
+    overwrite_from_index: bool = False,
+    overwrite_from_index_token: Optional[str] = None,
+    build_tags: Optional[List[str]] = None,
+    add_arches: Optional[Set[str]] = None,
+    binary_image_config: Optional[Dict[str, Dict[str, str]]] = None,
+    index_to_gitlab_push_map: Dict[str, str] = {},
+    used_fbc_fragment: bool = False,
+) -> None:
+    """
+    Add fbc fragments to an fbc index image.
+
+    :param list fbc_fragments: list of fbc fragments that need to be added to final FBC index image
+    :param int request_id: the ID of the IIB build request
+    :param str binary_image: the pull specification of the container image where the opm binary
+        gets copied from.
+    :param str from_index: the pull specification of the container image containing the index that
+        the index image build will be based from.
+    :param set add_arches: the set of arches to build in addition to the arches ``from_index`` is
+        currently built for; if ``from_index`` is ``None``, then this is used as the list of arches
+        to build the index image for
+    :param dict index_to_gitlab_push_map: the dict mapping index images (keys) to GitLab repos
```

Comment:
Why do we use `index_to_gitlab_push_map` here? I may have missed something, but I would expect this in some IIB config (index <-> gitlab map).

Reply:
The map is defined in API config. It gets passed to the handler in the function call.